### PR TITLE
TKNECO-27: Git Task Initial Steps

### PR DESCRIPTION
**Pull request by otaviof**
Initial changes to bootstrap the repository, integration and end-to-end testing.

### CODE REVIEW

**Comment by otaviof** [comment]
@vdemeester what do you think about the "ALL CAPS" inputs? Let's get this discussion started here too, please.

**Comment by vdemeester** [comment]
@otaviof 2 comments :
- any ways to *not* have git submodules ? (I am really not a fan)
- how would it look if we are not "obfuscating" the scripts part ? It makes the task a bit opaque and hard to debug / figure out ? Maybe this means having an image based of `git-init` with those scripts in ?
- We need to enable openshift-ci on it 👼🏼 I see there is a `make e2e` so it will be easy. Do you think we should enable it prior to the PR or after ? 😛

**Comment by otaviof** [comment]
> @otaviof 2 comments :
> 
>     * any ways to _not_ have git submodules ? (I am really not a fan)

Well, the objective is having the bats framework available for development and testing, in other words, we are adding a specific dependency to the project, which we want to have control of its version.

For instance, for Golang projects we would be adding the dependency on `go.mod`, but for a shell-script based repository we need to rely on a different method.

Knowing the git-submodules are around for many years already, I think it's a good place to declare the project's development/testing dependency. 

>     * how would it look if we are not "obfuscating" the scripts part ? It makes the task a bit opaque and hard to debug / figure out ? Maybe this means having an image based of `git-init` with those scripts in ?

When you set the `params.VERBOSE=true` the shell commands are all printed out, as the following example:

```
---> Phase: Collecting cloned repository information ('/workspace/output/')
---> Phase: Setting output workspace as safe directory ('/workspace/output')
+ phase 'Collecting cloned repository information ("/workspace/output/")'
+ echo '---> Phase: Collecting cloned repository information ("/workspace/output/")...'
+ cd /workspace/output/
+ phase 'Setting output workspace as safe directory ("/workspace/output")'
+ echo '---> Phase: Setting output workspace as safe directory ("/workspace/output")...'
+ git config --global --add safe.directory /workspace/output
+ git rev-parse HEAD
+ result_sha=9c74cbf874df6810e33651c3fb446fea3b19823f
+ git log -1 '--pretty=%ct'
+ result_committer_date=1678975608
+ phase 'Reporting last commit date "1678975608"'
---> Phase: Reporting last commit date "1678975608"'
+ echo '---> Phase: Reporting last commit date "1678975608"...'
+ printf '%s' 1678975608
+ phase 'Reporting parsed revision SHA "9c74cbf874df6810e33651c3fb446fea3b19823f"'
+ echo '---> Phase: Reporting parsed revision SHA "9c74cbf874df6810e33651c3fb446fea3b19823f"...'
---> Phase: Reporting parsed revision SHA '9c74cbf874df6810e33651c3fb446fea3b19823f'
+ printf '%s' 9c74cbf874df6810e33651c3fb446fea3b19823f
+ phase 'Reporting repository URL "https://github.com/tektoncd/community.git"'
---> Phase: Reporting repository URL 'https://github.com/tektoncd/community.git'
+ echo '---> Phase: Reporting repository URL "https://github.com/tektoncd/community.git"...'
+ printf '%s' https://github.com/tektoncd/community.git
+ exit 0
```

By injecting the scripts in any compatible image, we allow users to pick the most suitable for their use-case, and they can see what's going on as the example above. Also, saves us having to build and maintain those images.

Definitely, we will improve this approach later on, but I think it's a good way to start.

>     * We need to enable openshift-ci on it 👼🏼 I see there is a `make e2e` so it will be easy. Do you think we should enable it prior to the PR or after ? 😛

Indeed! Perhaps we want to have it in place before this PR is merged. Would be better start with the complete CI testing 🙏

File: templates/task.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  {{- .Values.stepTemplate.resources | toYaml | nindent 6 }}
+{{- end }}
+{{- if .Values.stepTemplate.securityContext }}
+    securityContext:
+  {{- .Values.stepTemplate.securityContext | toYaml | nindent 6 }}
+{{- end }}
+
+  steps:
+    - name: load-scripts
+      image: $(params.GIT_INIT_IMAGE)
```
Comment (vdemeester) [inline templates/task.yaml:156]: Note that in https://issues.redhat.com/browse/TKNECO-33 we want to explicitely remove this 🙃. One reason is.. support 😅 If we allow user to use whatever images they want, it's nice for them, but **impossible** for us to support them.

**Review by vdemeester — APPROVED** [review]
Sounds good. Long term we'll have to figure out / decide which `git-init` image to use (as to fully support the task, we will probably need a `ubi-8` based image. *But* as is, it's a good start already.

**Pull request merged**